packages/contracts/contracts/engine/V2/GenArt721Minter_PBAB.sol

```diff
@@ -178,24 +178,38 @@ contract GenArt721Minter_PBAB is ReentrancyGuard {
     /**
      * @notice Purchases a token from project `_projectId`.
      * @param _projectId Project ID to mint a token on.
+     * @param _maxPricePerToken Maximum price of token being allowed by the purchaser, no decimal places. Required if currency is ERC20.
+     * @param _currencyAddress Currency address of token.
      * @return _tokenId Token ID of minted token
      */
     function purchase(
-        uint256 _projectId
+        uint256 _projectId,
+        uint256 _maxPricePerToken,
+        address _currencyAddress
     ) public payable returns (uint256 _tokenId) {
-        return purchaseTo(msg.sender, _projectId);
+        return
+            purchaseTo(
+                msg.sender,
+                _projectId,
+                _maxPricePerToken,
+                _currencyAddress
+            );
     }
 
     /**
      * @notice Purchases a token from project `_projectId` and sets
      * the token's owner to `_to`.
      * @param _to Address to be the new token's owner.
      * @param _projectId Project ID to mint a token on.
+     * @param _maxPricePerToken Maximum price of token being allowed by the purchaser, no decimal places. Required if currency is ERC20.
```

Comment:
I think since this is a dual-purpose minter (both ETH and ERC20), this should probably always be required, and it should match`msg.value` if minting with ETH?
```suggestion
     * @param _maxPricePerToken Maximum price of token being allowed by the purchaser, no decimal places.```